app/domain/authentication/authn_k8s/execute_command_in_container.rb

```diff
@@ -0,0 +1,141 @@
+# frozen_string_literal: true
+
+require 'command_class'
+require 'uri'
+require 'websocket'
+require 'rubygems/package'
+
+require 'active_support/time'
+require 'websocket-client-simple'
+
+module Authentication
+  module AuthnK8s
+
+    ExecuteCommandInContainer ||= CommandClass.new(
+      dependencies: {
+        env:                           ENV,
+        websocket_client:              WebSocket::Client::Simple,
+        ws_client_event_handler_class: WebSocketClientEventHandler,
+        message_log_class:             MessageLog,
+        validate_message:              MessageLog::ValidateMessage.new,
+        logger:                        Rails.logger
+      },
+      inputs:       %i(k8s_object_lookup pod_namespace pod_name container cmds body stdin)
+    ) do
+
+      extend Forwardable
+      def_delegators :@k8s_object_lookup, :kube_client
+
+      DEFAULT_KUBE_EXEC_COMMAND_TIMEOUT = 5
+
+      def call
+        init_ws_client
+        add_websocket_event_handlers
+        wait_for_close_message
+        verify_channel_is_closed
+        verify_error_stream_is_empty
+        websocket_messages
+      end
+
+      private
+
+      def init_ws_client
+        ws_client
+        ws_client_event_handler
+      end
+
+      def ws_client
+        @ws_client ||= @websocket_client.connect(server_url, headers: headers)
+      end
+
+      def ws_client_event_handler
+        @ws_client_event_handler ||= @ws_client_event_handler_class.new(
+          ws_client:        ws_client,
+          stdin:            @stdin,
+          body:             @body,
+          pod_name:         @pod_name,
+          validate_message: @validate_message,
+          message_log:      @message_log_class.new,
+          logger:           @logger
+        )
+      end
+
+      def add_websocket_event_handlers
+        # We need to set this as if we'll use @ws_client_event_handler inside
+        # the curly brackets it will look for such a member in the websocket_client
+        # class
+        ws_client_event_handler = @ws_client_event_handler
```

Comment:
I think I tested this the other night and it was not needed.  I'll check again today.